### PR TITLE
new methods for commission documents 

### PR DESCRIPTION
**Pull request by duchenean**

### CODE REVIEW

File: src/Products/MeetingLalouviere/browser/overrides.py [hunk 1]
```diff
@@ -306,6 +306,35 @@ def get_categories_for_commission(self, commission_num):
...
+        """
+        meeting = self.context
+        premeetingdate = meeting.getPreMeetingDate()
+        return u"({weekday} {day} {month} {year} ({time}), {place})".format(
+            weekday=meeting.utranslate("weekday_%s" % premeetingdate.aDay().lower(), domain="plonelocales"),
+            day=premeetingdate.strftime('%d'),
+            month=meeting.translate('month_%s' % premeetingdate.strftime('%b').lower(), domain='plonelocales').lower(),
+            year=premeetingdate.strftime('%Y'),
+            time=premeetingdate.strftime('%HH%M'),
+            place=meeting.getPreMeetingPlace()
```
Comment (odelaere) [inline src/Products/MeetingLalouviere/browser/overrides.py]: There is more than one PreMeetingPlace. Please handle them all

**Comment by duchenean** [comment]
Thanks, I’ll take a look tonight and update this tomorrow :)

File: src/Products/MeetingLalouviere/browser/overrides.py [hunk 2]
```diff
@@ -308,6 +308,35 @@ def get_categories_for_commission(self, commission_num):
...
 
+    def get_commission_items(self, itemUids, commission_num=1, supplement=False):
+        """
+        Get the items of the commission
+        :param commission_num: number of the commission we want ([1-6])
+        :param supplement: supplement items
+        :return: list of meetingItem
+        """
+        cats = self.get_categories_for_commission(commission_num)
+        if supplement:  # If we want the supplements items
```
[the remark this answers is not in the document]
Comment (duchenean) [reply src/Products/MeetingLalouviere/browser/overrides.py]: Now there is one : type="*"

File: src/Products/MeetingLalouviere/browser/overrides.py [hunk 3]
```diff
@@ -308,6 +308,35 @@ def get_categories_for_commission(self, commission_num):
...
+            cats = [cat + '-1er-supplement' for cat in cats]  # append supplement suffix to the categories
+        return self.real_context.adapted().getPrintableItems(itemUids, categories=cats)
+
+    def format_premeetingdate(self):
+        """
+        format pre-meeting date like this : (Lundi 20 mai 2019 (18H30), Salle du Conseil communal)
+        :return: formatted pre-meeting date string
+        """
+        meeting = self.context
+        premeetingdate = meeting.getPreMeetingDate()
```
[the remark this answers is not in the document]
Comment (duchenean) [reply src/Products/MeetingLalouviere/browser/overrides.py]: Done

File: src/Products/MeetingLalouviere/browser/overrides.py (hunk 1 above)
Comment (duchenean) [reply src/Products/MeetingLalouviere/browser/overrides.py]: Done

File: src/Products/MeetingLalouviere/browser/overrides.py [hunk 4]
```diff
@@ -308,6 +308,35 @@ def get_categories_for_commission(self, commission_num):
             # single category as a string
             return [cat]
 
+    def get_commission_items(self, itemUids, commission_num=1, supplement=False):
```
[the remark this answers is not in the document]
Comment (duchenean) [reply src/Products/MeetingLalouviere/browser/overrides.py]: Done

File: src/Products/MeetingLalouviere/browser/overrides.py [hunk 5]
```diff
@@ -306,6 +306,57 @@ def get_categories_for_commission(self, commission_num):
...
+        :param supplement: supplement items
+        :param type: must be 'normal', 'supplement' or '*'
+        :return: list of meetingItem
+        """
+        cats = self.get_categories_for_commission(commission_num)
+        print cats
+        if type == 'supplement':  # If we want the supplements items only
+            cats = [cat + '-1er-supplement' for cat in cats]  # append supplement suffix to the categories
+        elif type == '*':  # If we want all items
+            cats = [cat + '-1er-supplement' for cat in cats] + cats
```
Comment (odelaere) [inline src/Products/MeetingLalouviere/browser/overrides.py]: Order should be usual categories first. Then all supplements in the exact same order than the usual

Comment (duchenean) [reply src/Products/MeetingLalouviere/browser/overrides.py]: Yep, done 
(hopefully, it's good :sweat_smile: )

**Pull request merged**